R/derive_vars_dy.R

```diff
@@ -0,0 +1,87 @@
+#' Derive Analysis Relative Day
+#'
+#' Adds the analysis relative day (`--DY`) to the dataset, i.e., study
+#' day of analysis date.
+#'
+#' @param dataset Input dataset
+#'
+#'   The columns specified by the `reference_date` and the `source_vars` parameter are
+#'   expected.
+#'
+#' @param reference_date The start date column, e.g., date of first treatment
+#'
+#'   A date or date-time object column is expected.
+#'
+#' @param source_vars A list of datetime or date variables created using `vars()` from
+#'   which dates are to be extracted.
+#'
+#' @author Teckla Akinyi
+#'
+#' @details The study day is derived as number of days from the reference date
+#'   to the end date. If it is nonnegative, one is added. I.e., the study day of the
+#'   reference date is 1.
+#'
+#' @return The input dataset with `--DY` corresponding to the `--DTM` or `--DT`
+#'     source variable(s) added
+#'
+#' @keywords derivation bds occds timing
```

Comment:
I would use `adam` instead of `bds` and `occds` because it could be used for any ADaM dataset.